docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+  branch = main # Ensure the branch is correct
+----
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+=== Onboard the project onto {ProductName}
+
+After creating the repository, onboard to {ProductName} and configure your application and its components.
+
+**Note:**
+The xref:/getting-started/index.adoc[Getting started docs] are currently geared towards configuring YAML files.
+____
+At the time of publication, to create applications in {ProductName}, you need to manually configure them by editing YAML files
+____
+
+This repo was configured using the {ProductName} UI.
+There does not seem to be any detailed steps on how to do this yet via the UI.
+
+Here are some **brief** steps on onboarding to {ProductName} using the UI:
+
+1. Create an application in your workspace.
+2. Configure your application component.
+3. If you're using Github, ensure that you have the link:https://github.com/apps/red-hat-konflux[Red Hat {ProductName} bot] enabled.
+4. Check your {ProductName} CI results. Ensure that your component was able to be pulled from, built and tested.
+
+For more information on configuring creating applications and components, please read xref:/how-tos/creating.adoc[this documentation].
+
+== Benefits of this pattern
+
+1. **Separation of Concerns**: The downstream repo can stay focused on its specific development goals while the submodule handles its own code. This keeps your downstream repo clean and modular.
+2. **Automatic Dependency Syncing with link:https://github.com/renovatebot/renovate[Renovate]**: By automating the dependency sync process, whether its syncing the git submodule, the Containerfile, etc. your downstream repo can always use the latest code from the upstream repo(s), ensuring that your project benefits from new features, bug fixes, and security updates.
+3. **Version Control for upstream repos**: You can lock the submodule to a specific commit or version, giving you control over when and how updates are integrated. This is especially useful if an upstream update introduces breaking changes.
+4. **{ProductName} CI**: Automated building and testing of your software artifacts help ensure that changes in the upstream repo don't break your downstream project.
+
+== Potential drawbacks
+
+1. **Complexity**: Managing submodules and automating their updates adds complexity to your workflow, such as:
+  - Debugging issues related to submodule updates
+  - Your automation configuration can be more challenging.
+2. **Dependency Management**: Breaking changes in the upstream repo can affect your downstream repo, requiring manual intervention to fix integration issues.
```

Comment:
One of the biggest challenges here is if the build process changes upstream and we have our own copy of the build configuration (i.e. Containerfile) in our repo. There are some specific practices that we can propose to mitigate that. Can we be more prescriptive here ... or at least more cautionary?

Reply:
I leaned more cautionary after editing this section, since I do not know what practices to prescribe yet. If you have any recommendations I will add them